File: naive/466.count-linked-list-nodes/Solution.go
package nodes

func main() {

}

type ListNode struct {
	Val  int
	Next *ListNode
}

/**
 * @param head: the first node of linked list.
 * @return: An integer
 */
func countNodes(head *ListNode) int {
	tmp := head
	count := 0
	for tmp != nil {
		count++
		tmp = tmp.Next
	}
	return count
}

func countNodesWorseEdition(head *ListNode) int {
	if head == nil {
		return 0
	}
	// write your code here
	count := 1
	for head.Next != nil {
		count++
		head.Next = head.Next.Next
	}
	return count
}

File: naive/466.count-linked-list-nodes/Solution.py
"""
Definition of ListNode
class ListNode(object):
    def __init__(self, val, next=None):
        self.val = val
        self.next = next
"""

class Solution:
    """
    @param head: the first node of linked list.
    @return: An integer
    """
    def countNode(self, head):
        count = 0
        while head:
            head = head.next
            count += 1
        return count
    

    
    def countNodes(self, head):
        # write your code here
        tmp = head
        count = 0
        while (tmp):
            count += 1
            tmp = tmp.next
        return count
    